src/Microsoft.Health.Fhir.Shared.Core/Features/Search/Parameters/SearchParameterEditor.cs

```diff
@@ -0,0 +1,48 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using EnsureThat;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Definition;
+using Microsoft.Health.Fhir.Core.Features.Definition.BundleWrappers;
+using Microsoft.Health.Fhir.Core.Features.Search.Registry;
+
+namespace Microsoft.Health.Fhir.Shared.Core.Features.Search.Parameters
+{
+    public class SearchParameterEditor : ISearchParameterEditor
```

Comment:
>Editor [](start = 32, length = 6)

DefinitionManager, DefinitionBuilder and Editor walk into the bar...

So there is StatusManager as well which looks like contains reference on DefinitionManager.

I can understand model where you have layer to access DB (DataStore)
There should be collection which contains all current search params and push updates to DB. (is it StatusManager? Or DefinitionManager?)

Builder I guess is helper class which builds that collection.

I'm sure there is reason for all this classes to exist, it just it's not clear from names and documentation aren't great. And that raises question, what this class does, and why it's not StatusManager?

Reply:
Good question :).  StatusManager, as well as DefinitionManager are in Core, while SearchParameterEditor is in Shared.Core.  Shared.Core is dependent on a specific version of FHIR and it's main job is grab the version specifc SearchParameter resource and put it into a bundle which is then wrapped in the version agnostic bundleWrapper.